packages/server/src/api/todo/adx/account.ts

```diff
@@ -67,6 +97,14 @@ export default function (server: Server) {
 
     await db.registerUser(email, username, did, password)
 
+    if (config.inviteRequired && inviteCode) {
+      const inviteCodeUse = new InviteCodeUse()
+      inviteCodeUse.code = inviteCode
+      inviteCodeUse.usedBy = did
+      inviteCodeUse.usedAt = new Date().toISOString()
+      await db.db.getRepository(InviteCodeUse).insert(inviteCodeUse)
```

Comment:
I think a double-use could technically occur with the invite code between the `useCount` check and this insert.  I don't think it needs to hold-up the PR, though— probably could take a pass at this and similar issues separately from this work.

Reply:
yup i thought of that as well. I'm banking on requests being sparse enough that that isn't an issue 😅 

there's serveral spots where writes should be transactional. and a couple like this where we need to make sure writes are consistent with the current DB value (updating the repo root for instance to prevent branched histories)

Reply:
I'm okay with leaving the bug if we leave a TODO, maybe create a tracking issue the areas we know transactionality is an issue

Reply:
good idea